noir/noir-repo/compiler/noirc_evaluator/src/brillig/brillig_ir/codegen_control_flow.rs

```diff
@@ -102,18 +73,14 @@ impl<F: AcirField + DebugToString> BrilligContext<F> {
         // Deallocate our temporary registers
         self.deallocate_single_addr(iterator_less_than_iterations);
         self.deallocate_single_addr(iterator_register);
-        // Only deallocate step if we allocated it
-        if step.is_none() {
-            self.deallocate_register(step_register);
-        }
```

Comment:
It shouldn't since step is either controlled by the caller (we shouldn't deallocate) or usize_one (we shouldn't deallocate either) the if before was because we were allocating within make_usize_constant